src/parser/length.rs

```diff
@@ -0,0 +1,71 @@
+use std::str::FromStr;
+
+use super::*;
+use crate::tokenizer::*;
+
+#[cfg(test)]
+mod tests;
+
+#[derive(Debug, PartialEq)]
+pub enum Distance {
+    Zero(),
+    Distance(f64, DistanceUnit),
+    Percentage(f64),
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum DistanceUnit {
+    Pixels,          // px
+    Centimeters,     // cm
+    Inches,          // in
+    Points,          // pt
+    FontSize,        // em
+    RootFontSize,    // rem
+    ViewportHeight,  // vh
+    ViewportWidth,   // vw
+    RootBlockSize,   // vb
+    RootInlineSize,  // vi
```

Comment:
Should probably start with "Viewpoint" for consistentcy

Reply:
So like `ViewpointRootBlockSize` and `ViewpointRootInlineSize` correct?

Reply:
I don't think you need "root" really